models/dataset.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+"""
```

Comment:
The docstring at the top of the file should also briefly describe what is in the file, see here for example:
https://github.com/geomstats/geomstats/blob/471a70578afe3926520a7fd8247e73ca05b387f5/tests/tests_geomstats/test_backends.py#L1